### PR TITLE
feat(Commerce Atomic): query suggest atomic component

### PR DESCRIPTION
**Pull request by y-lakhdar**
image

https://coveord.atlassian.net/browse/KIT-3128

### CODE REVIEW

File: packages/atomic-angular/projects/atomic-angular/src/lib/stencil-generated/components.ts [hunk 1]
```diff
@@ -161,6 +161,28 @@ export class AtomicCommerceInterface {
 export declare interface AtomicCommerceInterface extends Components.AtomicCommerceInterface {}
 
 
+@ProxyCmp({
```
Comment (olamothe) [inline packages/atomic-angular/projects/atomic-angular/src/lib/stencil-generated/components.ts:164]: Mark the atomic component `@internal` in JSDoc, and do a rebuild: This should clean up those generated files.

For the time being I would prefer we don't output angular/react until we're more mature with Atomic commerce.

File: packages/atomic/src/components/commerce/search-box-suggestions/atomic-commerce-search-box-query-suggestions/atomic-commerce-search-box-query-suggestions.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  QuerySuggestionText,
+} from '../../../common/suggestions/query-suggestions';
+import {
+  dispatchSearchBoxSuggestionsEvent,
+  SearchBoxSuggestionElement,
+  SearchBoxSuggestions,
+  SearchBoxSuggestionsBindings,
+} from '../../../common/suggestions/suggestions-common';
+
+/**
```
Comment (olamothe) [inline packages/atomic/src/components/commerce/search-box-suggestions/atomic-commerce-search-box-query-suggestions/atomic-commerce-search-box-query-suggestions.tsx:26]: Add `@internal` annotation to JSDoc

File: packages/atomic/src/components/commerce/search-box-suggestions/atomic-commerce-search-box-query-suggestions/atomic-commerce-search-box-query-suggestions.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
+import {
+  SearchBox,
+  Suggestion,
+  CommerceEngine,
+  loadQuerySuggestActions,
+} from '@coveo/headless/commerce';
+import {
```
Comment (olamothe) [inline packages/atomic/src/components/commerce/search-box-suggestions/atomic-commerce-search-box-query-suggestions/atomic-commerce-search-box-query-suggestions.tsx:7]: We must not do this kind of import, it's very problematic. 😄  

I realize this is a copy paste from classic search box suggestion component, which I did not catch while doing my refactor. But we should not spread it more.

**Review by olamothe — APPROVED** [review]
Progressing well !

**Review by louis-bompart — APPROVED** [review]
Other than @olamothe comments, LGTM.

File: packages/atomic/src/components/commerce/search-box-suggestions/atomic-commerce-search-box-query-suggestions/atomic-commerce-search-box-query-suggestions.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+} from '../../../common/suggestions/query-suggestions';
+import {
+  dispatchSearchBoxSuggestionsEvent,
+  SearchBoxSuggestionElement,
+  SearchBoxSuggestions,
+  SearchBoxSuggestionsBindings,
+} from '../../../common/suggestions/suggestions-common';
+
+/**
+ * The `atomic-commerce-search-box-query-suggestions` component can be added as a child of an `atomic-search-box` component, allowing for the configuration of query suggestion behavior.
```
Comment (fpbrault) [inline packages/atomic/src/components/commerce/search-box-suggestions/atomic-commerce-search-box-query-suggestions/atomic-commerce-search-box-query-suggestions.tsx:23]:
```suggestion
 * The `atomic-commerce-search-box-query-suggestions` component can be added as a child of an `atomic-commerce-search-box` component, allowing for the configuration of query suggestion behavior.
```

**Review by fbeaudoincoveo — APPROVED** [review]

File: packages/headless/src/features/commerce/query-suggest/query-suggest-actions-loader.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  FetchQuerySuggestionsActionCreatorPayload,
+  QuerySuggestActionCreators,
+} from '../../query-suggest/query-suggest-actions-loader';
+import {querySuggestReducer as querySuggest} from '../../query-suggest/query-suggest-slice';
+import {fetchQuerySuggestions} from './query-suggest-actions';
+
+/**
+ * The query suggest action creators for commerce.
+ */
+export interface CommerceQuerySuggestActionCreators
```
Comment (fpbrault) [inline packages/headless/src/features/commerce/query-suggest/query-suggest-actions-loader.ts:24]: I noticed this is prefixed with "Commerce" but `RecentQueriesActionCreators` isn't

**Pull request merged**